src/blocks/utils/ImageWithModal.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { Dialog, IconButton, DialogContent } from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import SwiperCore from 'swiper';
+import { Navigation, Zoom } from 'swiper/modules';
+
+import 'swiper/css';
+import 'swiper/css/navigation';
+import 'swiper/css/zoom';
+
+// Initialize Swiper modules
+SwiperCore.use([Navigation, Zoom]);
+
+interface ImageWithModalProps {
+  urls: string[]; // Array of image source URLs
+}
+
+const ImageWithModal: React.FC<ImageWithModalProps> = ({ urls }) => {
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = (event: any) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setOpen(true);
+  };
+  const handleClose = () => setOpen(false);
+
+  return (
+    <>
+      <style>
+        {`
+          .hide-navigation-buttons .swiper-button-next,
+          .hide-navigation-buttons .swiper-button-prev {
+            display: none;
+          }
+        `}
+      </style>
+      <div style={{ display: 'flex', cursor: 'pointer', gap: '10px' }} onClick={(e) => handleOpen(e)}>
+        {urls.map((src, index) => ( // conditional styling for 1 image vs many
+          <img key={index} src={src} style={{ width: urls.length === 1 ? "100%" : "auto", maxWidth: '300px', height: urls.length === 1 ? "auto" : "300px", objectFit: 'cover' }} alt="Thumbnail" />
+        ))}
+      </div>
+      {/* full screen image modal */}
```

Comment:
great example here for other miniapps that want to go fullscreen, we can write a tutorial for it